src/core/thread/data_poll_manager.cpp

```diff
@@ -353,20 +353,21 @@ void DataPollManager::SendFastPolls(uint8_t aNumFastPolls)
 
 void DataPollManager::ScheduleNextPoll(PollPeriodSelector aPollPeriodSelector)
 {
+    uint32_t now;
+
     if (aPollPeriodSelector == kRecalculatePollPeriod)
     {
         mPollPeriod = CalculatePollPeriod();
     }
 
-    if (mTimer.IsRunning())
-    {
-        mTimer.StartAt(mTimerStartTime, mPollPeriod);
-    }
-    else
+    now = TimerMilli::GetNow();
+
+    if (!mTimer.IsRunning() || now >= (mTimerStartTime + mPollPeriod))
```

Comment:
Not sure about comparing times with `>=` due to it wrapping around (it's `uint32_t`).  Every ~50 days the milli second `uint32_t` time wraps to zero and then this comparison would be incorrect. 
We have `TimerScheduler::IsStrictlyBefore` which takes care of this (the key is to compare difference not absolute times).